### PR TITLE
[dsymbolsem] Tweak punctuation for 3 error messages

### PR DESCRIPTION
**Pull request by ntrel**
Fix Issue 13656 - clarify error when trying to declare a variable of type ref. Mention functions can be ref.
Tweak 2 inout errors.

I moved inout tests from failinout3748a.d and failinout3748b.d as I noticed their tests fit well in existing files with the same error messages. Best not to split closely related tests across files with only one or two error messages in each.

### CODE REVIEW

File: compiler/src/dmd/dsymbolsem.d [hunk 1]
```diff
@@ -835,7 +835,7 @@ private extern(C++) final class DsymbolSemanticVisitor : Visitor
                 }
                 if (!isWild)
                 {
-                    dsym.error("`inout` variables can only be declared inside `inout` functions");
+                    dsym.error(": `inout` variables can only be declared inside `inout` functions");
```
Comment (dkorpel) [inline compiler/src/dmd/dsymbolsem.d:838]: You could also use the global `.error` function and pass it `dsym.toChars()` so you can put the name in a more convenient part of the message

Comment (ntrel) [reply compiler/src/dmd/dsymbolsem.d:838]: I think there are a lot of these still to fix (for another pull), so I'd rather keep the changes simple.

Comment (ntrel) [reply compiler/src/dmd/dsymbolsem.d:838]: I think there are a lot of these still to fix (for another pull), so I'd rather keep the changes simple.

**Review by RazvanN7 — APPROVED** [review]
Rebasing to latest master will fix the CI errors

File: compiler/src/dmd/dsymbolsem.d [hunk 2]
```diff
@@ -544,7 +544,7 @@ private extern(C++) final class DsymbolSemanticVisitor : Visitor
             }
         }
         if ((dsym.storage_class & STC.auto_) && !inferred)
-            dsym.error("storage class `auto` has no effect if type is not inferred, did you mean `scope`?");
+            dsym.error("- storage class `auto` has no effect if type is not inferred, did you mean `scope`?");
```
Comment (ntrel) [inline compiler/src/dmd/dsymbolsem.d:547]: I used a colon first but it gets formatted with a space before the colon, so I think a hyphen looks better.

**Pull request merged**